### PR TITLE
feat(github): add organization domain verification check

### PR DESCRIPTION
**Pull request by kushpatel321**
Fixes Issue: #8663

This pull request adds a new check to Prowler that verifies whether a GitHub organization has a verified badge. This badge shows that GitHub has confirmed the organization's identity, which helps users trust the organization is legitimate and not impersonating anyone. The new check uses the GitHub API to check if an organization has the "is_verified" field set to true and includes this result in the Prowler scan output.

Steps to review:
Look at the organization_verified_badge.py. This file contains the logic for checking if an organization is verified.
Review the metadata file organization_verified_badge.metadata.json to make sure it follows the Prowler project’s guidelines.
Ensure the check properly shows up in the scan results and that it adds to the total check count.

### CODE REVIEW

File: prowler/compliance/github/cis_1.0_github.json [hunk 1]
```diff
@@ -778,7 +778,9 @@
     {
       "Id": "1.3.9",
       "Description": "Confirm the domains an organization owns with a \"Verified\" badge.",
-      "Checks": [],
+      "Checks": [
+        "organization_domain_verified"
```
Comment (andoniaf) [inline prowler/compliance/github/cis_1.0_github.json]:
```suggestion
        "organization_verified_badge"
```

This field needs to match the check name.

File: prowler/providers/github/services/organization/organization_verified_badge/organization_verified_badge.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            organization_client,
+        )
+
+        findings = []
+
+        for org in organization_client.organizations.values():
+            report = CheckReportGithub(metadata=self.metadata(), resource=org)
+
+            is_verified = False
+            if hasattr(org, "raw_data") and isinstance(org.raw_data, dict):
```
Comment (andoniaf) [inline prowler/providers/github/services/organization/organization_verified_badge/organization_verified_badge.py]: The check accesses raw_data dict instead of using the already-available `org.is_verified` field from the typed `Org` model.

File: prowler/providers/github/services/organization/organization_verified_badge/organization_verified_badge.metadata.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{
+  "Provider": "github",
+  "CheckID": "organization_verified_badge",
+  "CheckTitle": "GitHub organization has a verified badge",
+  "CheckType": [],
+  "ServiceName": "organization",
+  "SubServiceName": "",
+  "ResourceIdTemplate": "arn:github:organization::{OrganizationName}",
```
Comment (andoniaf) [inline prowler/providers/github/services/organization/organization_verified_badge/organization_verified_badge.metadata.json]:
```suggestion
  "ResourceIdTemplate": "",
```
This field is in the process of being [deprecated](https://docs.prowler.com/developer-guide/checks#resourceidtemplate) and should be left empty.

File: prowler/providers/github/services/organization/organization_verified_badge/organization_verified_badge.metadata.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  "Severity": "medium",
+  "ResourceType": "GitHubOrganization",
+  "Description": "Checks whether a GitHub organization has a verified badge.",
+  "Risk": "Unverified organizations may be easier to impersonate, increasing the risk of phishing or trust abuse.",
+  "RelatedUrl": "https://docs.github.com/en/organizations/managing-organization-settings/verifying-or-approving-a-domain-for-your-organization",
```
Comment (andoniaf) [inline prowler/providers/github/services/organization/organization_verified_badge/organization_verified_badge.metadata.json]:
```suggestion
  "RelatedUrl": "",
```

[Deprecated](https://docs.prowler.com/developer-guide/checks#relatedurl). Use AdditionalURLs for adding your URLs references.

**Review by andoniaf — CHANGES_REQUESTED** [review]
Thanks for the contribution @kushpatel321! 🖖

Beyond the inline comments, there are a few more things to address before we can merge:

###  Tests (required)

  This check needs unit tests. You can follow the pattern in `organization_members_mfa_required_test.py` as a reference. At minimum, please add test cases for:

  - No organizations (empty dict)
  - Organization with is_verified=True (PASS)
  - Organization with is_verified=False (FAIL)
  - Organization with is_verified=None (edge case)

  The test file should live at:
  `tests/providers/github/services/organization/organization_verified_badge/organization_verified_badge_test.py`

###  CHANGELOG entry (required)

  Please add an entry in prowler/CHANGELOG.md under ## [Unreleased] → ### Added for the new check.

### Metadata improvements
  - Remediation → Other: Currently empty. It would be helpful to include the steps to verify a domain through GitHub's UI (Organization Settings → Verified & approved domains → Add a domain → follow DNS verification steps).
  - Consider adding "trust-boundaries" alongside "identity-access" in the Categories array, since this check is fundamentally about organizational trust and authenticity.

Let me know if you have any questions!

**Review by kushpatel321 — COMMENTED** [review]
review changes made

**Review by andoniaf — DISMISSED** [review]

**Review by andoniaf — APPROVED** [review]
